vehicle/carwings.go

```diff
@@ -58,46 +57,30 @@ func NewCarWingsFromConfig(other map[string]interface{}) (api.Vehicle, error) {
 		log:      log,
 		user:     cc.User,
 		password: cc.Password,
-		region:   cc.Region,
+		session:  &carwings.Session{Region: cc.Region},
 	}
 
 	v.statusG = provider.NewCached(func() (bool, error) {
 		return true, v.status()
 	}, cc.Cache).BoolGetter()
-	return v, nil
-}
 
-// Init new carwings session
-func (v *CarWings) initSession() error {
-	v.session = &carwings.Session{
-		Region: v.region,
-	}
-	return v.session.Connect(v.user, v.password)
+	return v, nil
 }
 
-func (v *CarWings) status() (err error) {
-	if v.session == nil {
-		if err = v.initSession(); err != nil {
-			return api.ErrNotAvailable
-		}
-	}
-
-	var bs carwings.BatteryStatus
-	if bs, err = v.session.BatteryStatus(); err == nil {
+func (v *CarWings) status() error {
+	bs, err := v.session.BatteryStatus()
+	if err == nil {
 		if elapsed := time.Since(bs.Timestamp); elapsed > carwingsStatusExpiry {
-
+			// api result is stale
 			if v.refreshKey != "" {
 				return v.refreshResult()
 			}
 
 			if err = v.refreshRequest(); err != nil {
 				return err
 			}
-			// check result in next retry
+
 			err = api.ErrMustRetry
-		} else {
-			// Update finished in the meantime, reset key for next request
-			v.refreshKey = ""
```

Comment:
Ich habe Fälle gesehen, in denen das Ergebnis zwischenzeitlich angekommen ist und elapsed dann nicht mehr über dem ExpiryTimeout war. Dann wurde der refreshKey nicht gelöscht und somit beim nächsten durchlauf erst nach einem veralteten Key gesucht. Ich versuche gerade das nochmal mit dem aktuellen Stand zu reproduzieren,